### ISSUE TITLE
Kernel keeps disconnecting every 1-5 minutes. No code will output in Jupyter Notebook..

### ISSUE THREAD

**Issue by ThomasKirkatrick** [jupyter/notebook#3417]
Every 1-5 minutes or so my Kernel in my Jupyter  notebook is disconnecting. It shows a loading symbol but never reconnects. I have tried restarting my kernels, but the problem is persisting and I cannot write any new code or continue with any future lectures until the problem is solved.

My terminal reads:

[I 23:07:59.310 NotebookApp] Adapting to protocol v5.1 for kernel d1955ea9-de13-4bff-bbf4-147cac52b7eb

[I 23:07:59.311 NotebookApp] Restoring connection for d1955ea9-de13-4bff-bbf4-147cac52b7eb:72dab206bd084aecba35dc1f4070c5c3

**Comment by takluyver** [comment]
Are you running Jupyter locally, or accessing it on a server running elsewhere?

If locally, do you have any security software installed? We've had problems with some security software blocking our connections.

If you haven't already, try with a different browser from the one you normally use.

**Comment by ThomasKirkatrick** [comment]
I believe I am running it locally. Is there a definitive way to check? I do
have some Malware software installed on my system. Any advice on how to fix
the issue?

**Comment by takluyver** [comment]
If the URL in the browser is something like `localhost:8888` or `127.0.0.1:8888`, then it's local.

I'm assuming you mean anti-malware software. If you've got malware, that's bad ;-). Try stopping/disabling it temporarily to see if that's what's causing the problem

**Comment by ThomasKirkatrick** [comment]
Yes anti-malware haha.. It does appear it is local. When I am finished with
work and continue to mess with my capstone project I will give it a shot.
If I continue to experience problems I will update you! If everything goes
smooth and we don't speak, have a great rest of your week.

thanks for the quick reply,

Thomas

**Comment by ThomasKirkatrick** [comment]
I am still experiencing the problem. Even with my firewall down. It is only
when I run certain lines of code.

Thomas

**Comment by takluyver** [comment]
> It is only when I run certain lines of code.

That sounds like a good clue. What lines of code cause it?

**Comment by ThomasKirkatrick** [comment]
This line of code seems to hang up my notebook.

player1_marker , player2_marker = player_input()

Here is my player_input() function:

def player_input():

    marker = ''

    # Keep asking Player 1 to choose X or O

    while not (marker == 'X' or marker == 'O'):
        marker = input('Player 1, choose X or O: ').upper()

    if marker == 'X':

        return ('X','O')
    else:
        return ('O','X')

**Comment by takluyver** [comment]
That's weird, I don't know why that would make it disconnect. How far does it get? Do you see the input box in the notebook to type in an answer?

**Comment by ThomasKirkatrick** [comment]
Yes I see an input box. I can use the box multiple times. It just always
eventually crashes after using inputs in my code. All my homework’s I
experience no issues with kernel connections. I start experiencing issues
when I have done my two projects on Udemy, both required input functions.
The kernel disconnections always “seems” to occur when I get the the
portion that requires the input functions. Could also be because there are
a higher number of lines being run? Any advice would be greatly appreciated.

Thomas

**Comment by takluyver** [comment]
Is the issue with something running locally, or with a notebook server hosted on Udemy?

**Comment by ThomasKirkatrick** [comment]
Udemy is just a video I watch and follow along. It is local in my Jupyter
notebook.

**Comment by AnranWang** [comment]
> Are you running Jupyter locally, or accessing it on a server running elsewhere?
> 
> If locally, do you have any security software installed? We've had problems with some security software blocking our connections.
> 
> If you haven't already, try with a different browser from the one you normally use.

i have the same proplem when i run it on a server, do you have a clue of it ?

**Comment by ThomasKirkatrick** [comment]
None, I still experience it from time to time with large code.

**Comment by samstelzner** [comment]
I seem to be experiecing this same issue (while trying to run code on Jupyter Notebook as prompted by Udemy course). Kernel just runs on even the most basic functions. No output.

In[*]: def spy_game(nums):
    return "SAM"
            
# Check
In[*] spy_game([1,2,4,0,0,7,5])

**Comment by sam2332** [comment]
i am having this issue and its super annoying, i have a long running script to pull image data from a camera. even if i leave the window open it will eventually say that there are no running cells even though its while True with a sleep.

this is on a remote server using password auth.

i have a sneaking suspision it has to do with if you open a new window to the workspace and leave both open

**Comment by Aanyajain** [comment]
after 10 sec only jupyter notebook stops working like the symbol on top just breaks,after that no cmd works.Pls help anyone

**Comment by kevin-bates** [comment]
Hi @Aanyajain - What is displayed on the notebook server's console?  (please scrape, not screenshot)

**Comment by Aanyajain** [comment]
Actually nothing is coming,it is just showing star means in progress and after that nothing no output for like hours for just a print statement.

**Comment by kevin-bates** [comment]
It sounds like your kernel is dying.  There should be some output on the terminal window from which the `jupyter notebook` command was invoked.  Could you please provide that content?

**Comment by Aanyajain** [comment]
ya it just shows some get cmd and then buffering thats all nothing else i
guess

**Comment by kevin-bates** [comment]
Could you please scrape the contents of that window prior to where the buffering starts?  Looking for any kind of traceback or kernel specific messages.

**Comment by jonaszubindu** [comment]
Hi, I experience the same problem, but only if I want to store my jupyter notebooks. It doesn't matter how much output I have, just as soon as I want to store it after a longer time running it, the connection keeps failing every minute with "zmq message arrived on closed channel" and WebSocket ping timeout...", "Starting buffering for ..." It's very annoying because it almost always happens when I have to run any code for longer time. And the server does not reach capacity limits at all.

**Comment by grafke** [comment]
+1 
Using tornado 6.2, notebook 6.5.2
I constantly have to click on Kernel->Reconnect after having left the tab open for just a couple of minutes.

**Comment by NovaSenpai** [comment]
Opening using anaconda prompt and after opening it locally, closing the anaconda prompt cause this problem. Keep the anaconda prompt open and it will work perfectly. I know it is unusual, but we are using it locally using the support of anaconda prompt. Hope this helps.
